services/services.go

```diff
@@ -23,11 +23,18 @@ type Service struct {
 	Tags             []string    `mapstructure:"tags"`
 	Timeout          string      `mapstructure:"timeout"`
 	IPAddress        string
+	ConsulConfig     *ConsulConfig `mapstructure:"consul"`
 	healthCheckCmd   *commands.Command
 	discoveryService discovery.ServiceBackend
 	definition       *discovery.ServiceDefinition
 }
 
+// ConsulConfig handles additional Consul configuration.
+type ConsulConfig struct {
+	EnableTagOverride              bool   `mapstructure:"enabletagoverride"`
+	DeregisterCriticalServiceAfter string `mapstructure:"deregistercriticalserviceafter"`
```

Comment:
I feel like these should probably be camelCase, given that we use it elsewhere (ex. `preStart`)